modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/consensus/ConsensusState.scala

```diff
@@ -1,111 +1,81 @@
 
 import scala.concurrent.duration.FiniteDuration
 
-import org.tessellation.node.shared.infrastructure.consensus.declaration.kind.{PeerDeclarationKind, _}
-import org.tessellation.node.shared.infrastructure.consensus.trigger.ConsensusTrigger
+import org.tessellation.node.shared.infrastructure.consensus.declaration.PeerDeclaration
 import org.tessellation.schema.peer.PeerId
 import org.tessellation.security.hash.Hash
-import org.tessellation.security.signature.Signed
 
 import derevo.cats.{eqv, show}
 import derevo.circe.magnolia.{decoder, encoder}
 import derevo.derive
+import monocle.Lens
+import monocle.macros.GenLens
+
+@derive(eqv, encoder, decoder)
+case class Facilitators(value: List[PeerId])
+
+@derive(eqv, encoder, decoder)
+case class RemovedFacilitators(value: Set[PeerId])
+object RemovedFacilitators {
+  def empty: RemovedFacilitators = RemovedFacilitators(Set.empty)
+}
+
+@derive(eqv, encoder, decoder)
+case class WithdrawnFacilitators(value: Set[PeerId])
+object WithdrawnFacilitators {
+  def empty: WithdrawnFacilitators = WithdrawnFacilitators(Set.empty)
+}
+
+@derive(eqv, encoder, decoder, show)
+case class Candidates(value: Set[PeerId])
+object Candidates {
+  def empty: Candidates = Candidates(Set.empty)
+}
 
 @derive(eqv)
-case class ConsensusState[Key, Artifact, Context](
+case class ConsensusState[Key, Status, Outcome, Kind](
   key: Key,
-  lastOutcome: ConsensusOutcome[Key, Artifact, Context],
-  facilitators: List[PeerId],
-  status: ConsensusStatus[Artifact, Context],
+  lastOutcome: Outcome,
+  facilitators: Facilitators,
+  status: Status,
   createdAt: FiniteDuration,
-  removedFacilitators: Set[PeerId] = Set.empty,
-  withdrawnFacilitators: Set[PeerId] = Set.empty,
-  lockStatus: LockStatus = Open,
-  spreadAckKinds: Set[PeerDeclarationKind] = Set.empty
+  removedFacilitators: RemovedFacilitators = RemovedFacilitators.empty,
+  withdrawnFacilitators: WithdrawnFacilitators = WithdrawnFacilitators.empty,
+  lockStatus: LockStatus = LockStatus.Open,
+  spreadAckKinds: Set[Kind] // = Set.empty
 )
 
 object ConsensusState {
-  implicit def showInstance[K: Show, A, C]: Show[ConsensusState[K, A, C]] = { cs =>
+  implicit def showInstance[K: Show, S: Show, O, Kind: Show]: Show[ConsensusState[K, S, O, Kind]] = { cs =>
     s"""ConsensusState{
        |key=${cs.key.show},
        |lockStatus=${cs.lockStatus.show},
-       |facilitatorCount=${cs.facilitators.size.show},
-       |removedFacilitators=${cs.removedFacilitators.show},
-       |withdrawnFacilitators=${cs.withdrawnFacilitators.show},
+       |facilitatorCount=${cs.facilitators.value.size.show},
+       |removedFacilitators=${cs.removedFacilitators.value.show},
+       |withdrawnFacilitators=${cs.withdrawnFacilitators.value.show},
        |spreadAckKinds=${cs.spreadAckKinds.show},
        |status=${cs.status.show}
        |}""".stripMargin.replace(",\n", ", ")
   }
 
-  implicit class ConsensusStateOps[K, A, C](value: ConsensusState[K, A, C]) {
```

Comment:
The declaration kind and consensus status are now customized by every consensus, so these mappers needed to be implemented elsewhere.